src/read/dwarf.rs

```diff
@@ -1048,7 +1048,7 @@ impl<R: Reader> DwarfPackage<R> {
                 SectionId::DebugMacro | SectionId::DebugMacinfo => {
                     // These are valid but we can't parse these yet.
                 }
-                _ => return Err(Error::UnknownIndexSection),
+                _ => return Err(Error::UnknownSection(section.section)),
```

Comment:
Would it be better to introduce a new `enum IndexSectionId` which has only the kinds that are allowed in a `UnitIndex`?

Reply:
Maybe. It would avoid this possibility, and most consumers won't be directly using this section id so it won't affect them. Something for another PR though. I'd probably call it `DwoSectionId`.